plugins/@grouparoo/salesforce/src/lib/export/salesforceExporter.ts

```diff
@@ -43,24 +44,25 @@ const findAndSetDestinationIds: BatchMethodFindAndSetDestinationIds = async ({
 }) => {
   // search for these using the foreign key
   const { recordObject, recordMatchField, cacheData } = config.data;
-  const idType = "Id";
-
   const foreignKeyType = await getForeignKeyType(
     client,
     cacheData,
     recordObject,
     recordMatchField
   );
-
-  const parsedForeignKeys = parseForeignKeys(foreignKeys, foreignKeyType);
-
-  const query = { [recordMatchField]: parsedForeignKeys };
-  const fields = [idType, recordMatchField];
+  let validForeignKeys = parseForeignKeys(foreignKeys, foreignKeyType);
+  let fields = [idType, recordMatchField];
+  if (idType === recordMatchField) {
+    validForeignKeys = foreignKeys.filter((foreignKey) =>
+      VALID_ID_LENGTHS.includes(foreignKey.length)
+    );
+    fields = [idType];
```

Comment:
This looks to be the main change. What happens if we don't filter above. Does the whole query freak of if there is one that is not length 15 or 18? Where did we determine the 15 or 18?

Reply:
We can determine the length based on [this](https://help.salesforce.com/s/articleView?id=000324087&type=1)

Reply:
Yes, when there is an invalid ID on the query the entire query seems to freak out.

`warning: ExportError  Destination: accounts_destination_dsc_enrich - Salesforce Accounts DSC Enrich - Record: Unknown error=INVALID_QUERY_FILTER_OPERATOR: 
SELECT Id FROM Account WHERE Id IN ('sfabc')
                                                            ^
ERROR at Row:1:Column:30
`

Reply:
So if the `id` gets left off here, they there will be an error on that export saying what? I'm thinking we shoudl add an explicit error on that bad export here.

Reply:
So instead of 
```
validForeignKeys = foreignKeys.filter((foreignKey) =>
      VALID_ID_LENGTHS.includes(foreignKey.length)
    );
  ```
  something like this:
  ```
  const passed = [];
for (const id of foreignKeys) {
  if(VALID_ID_LENGTHS.includes(foreignKey.length)) {
      passed.push(foreignKey);
  }
  else {
      const found = getByForeignKey(foreignKey);
      if(found) {
            found.error = new Error(`Invalid Salesforce id length ${foreignKey`)
      }
  }
}
validForeignKeys = passed;
```